drivers/video/ov7670.c

```diff
@@ -0,0 +1,488 @@
+	{OV7670_MANU, 0x11},  /* Manual U Value */
+	{OV7670_MANV, 0xFF},  /* Manual V Value */
+	/* config the output window data, this can be configed later */
+	{OV7670_HSTART, 0x16}, /*  HSTART */
+	{OV7670_HSTOP, 0x04},  /*  HSTOP */
+	{OV7670_VSTRT, 0x02},  /*  VSTRT */
+	{OV7670_VSTOP, 0x7a},  /*  VSTOP */
+	{OV7670_HREF, 0x80},   /*  HREF */
+	{OV7670_VREF, 0x0a},   /*  VREF */
+
+	/* DCW Control, */
+	{OV7670_SCALING_DCWCTR, 0x11},
+
+	/* AGC/AEC - Automatic Gain Control/Automatic exposure Control */
+	{OV7670_GAIN, 0x00},  /*  AGC */
+	{OV7670_AECHH, 0x3F}, /* Exposure Value */
+	{OV7670_AECH, 0xFF},
+	{OV7670_COM8, 0x66},
+	{OV7670_COM9, 0x21}, /*  limit the max gain */
+	{OV7670_AEW, 0x75},
+	{OV7670_AEB, 0x63},
+	{OV7670_VPT, 0xA5},
+	/* Automatic white balance control */
+	{OV7670_AWBC1, 0x14},
+	{OV7670_AWBC2, 0xf0},
+	{OV7670_AWBC3, 0x34},
+	{OV7670_AWBC4, 0x58},
+	{OV7670_AWBC5, 0x28},
+	{OV7670_AWBC6, 0x3a},
+	/* Matrix Coefficient */
+	{OV7670_MTX1, 0x80},
+	{OV7670_MTX2, 0x80},
+	{OV7670_MTX3, 0x00},
+	{OV7670_MTX4, 0x22},
+	{OV7670_MTX5, 0x5e},
+	{OV7670_MTX6, 0x80},
+	/* AWB Control */
+	{0x59, 0x88},
+	{0x5a, 0x88},
+	{0x5b, 0x44},
+	{0x5c, 0x67},
+	{0x5d, 0x49},
+	{0x5e, 0x0e},
+	{0x6c, 0x0a},
+	{0x6d, 0x55},
+	{0x6e, 0x11},
+	{0x6f, 0x9f},
+	/* Lens Correction Option */
+	{OV7670_LCC1, 0x00},
+	{OV7670_LCC2, 0x00},
+	{OV7670_LCC3, 0x04},
+	{OV7670_LCC4, 0x20},
+	{OV7670_LCC5, 0x05},
+	{OV7670_LCC6, 0x04}, /* effective only when LCC5[2] is high */
+	{OV7670_LCC7, 0x08}, /* effective only when LCC5[2] is high */
+	/* Gamma Curve, needn't config */
+	{OV7670_SLOP, 0x20},
+	{OV7670_GAM1, 0x1c},
+	{OV7670_GAM2, 0x28},
+	{OV7670_GAM3, 0x3c},
+	{OV7670_GAM4, 0x55},
+	{OV7670_GAM5, 0x68},
+	{OV7670_GAM6, 0x76},
+	{OV7670_GAM7, 0x80},
+	{OV7670_GAM8, 0x88},
+	{OV7670_GAM9, 0x8f},
+	{OV7670_GAM10, 0x96},
+	{OV7670_GAM11, 0xa3},
+	{OV7670_GAM12, 0xaf},
+	{OV7670_GAM13, 0xc4},
+	{OV7670_GAM14, 0xd7},
+	{OV7670_GAM15, 0xe8},
+	/* Histogram-based AEC/AGC Control */
+	{OV7670_HAECC1, 0x78},
+	{OV7670_HAECC2, 0x68},
+	{OV7670_HSYEN, 0xff},
+	{0xa1, 0x03},
+	{OV7670_HAECC3, 0xdf},
+	{OV7670_HAECC4, 0xdf},
+	{OV7670_HAECC5, 0xf0},
+	{OV7670_HAECC6, 0x90},
+	/* Automatic black Level Compensation */
+	{0xb0, 0x84},
+	{0xb1, 0x0c},
+	{0xb2, 0x0e},
+	{0xb3, 0x82},
+	{0xb8, 0x0a},
+};
+
+static int ov7670_get_caps(const struct device *dev, enum video_endpoint_id ep,
+			   struct video_caps *caps)
+{
+	caps->format_caps = fmts;
+	return 0;
+}
+
+static int ov7670_set_fmt(const struct device *dev, enum video_endpoint_id ep,
+			  struct video_format *fmt)
+{
+	const struct ov7670_config *config = dev->config;
+	struct ov7670_data *data = dev->data;
+	uint8_t com7 = 0U;
+	uint8_t i = 0U;
+
+	if (fmt->pixelformat != VIDEO_PIX_FMT_RGB565 && fmt->pixelformat != VIDEO_PIX_FMT_YUYV) {
+		LOG_ERR("Only RGB565 and YUYV supported!");
+		return -ENOTSUP;
+	}
+
+	if (!memcmp(&data->fmt, fmt, sizeof(data->fmt))) {
+		/* nothing to do */
+		return 0;
+	}
```

Comment:
Yeah, it is a performance optimization. Other drivers in tree have the same code, so I figured it would make sense to include the feature here as well

Reply:
Thank you this makes sense!

Not to self: also add it to https://github.com/zephyrproject-rtos/zephyr/pull/73013